### DISCUSSION TITLE
Can't send a PATCH/PUT request with body via Fetch API

### DISCUSSION THREAD

**Discussion by melianmiko** [zepp-health/zeppos-forum#169] [Developer Feedback]
Hi,

I'm trying to make a request to some API from side-app with something like this:

```javascript
const resp = await fetch({
  url: "...",
  method: "PATCH",
  headers: {...},
  body: JSON.stringify({...})
});
```

And it works fine in simulator, but on real device, it looks like Zepp application didn't send request body to server. When using POST method, all works fine, but any else don't.

As temporary solution I've created a proxy-server that changes request method, but would be great if someone fix that =)

**Comment by htoooth** [comment]
@melianmiko  
我们去检查一下这个情况。你也可以确认一下是不是有可能是服务器没有设置正确。https://developer.mozilla.org/en-US/docs/Glossary/CORS

Let's check this situation. You can also confirm if it's possible that the server is not set up correctly. https://developer.mozilla.org/en-US/docs/Glossary/CORS

**Reply by melianmiko** [reply]
> Let's check this situation. You can also confirm if it's possible that the server is not set up correctly.

Of couse CORS is configured correctly, I got a response from server. But, I got an error, because request body isn't present. POST requests to the same server works fine from ZeppOS device.

**Reply by htoooth** [reply]
@melianmiko 

我检查了一下相关功能，目前 fetch 现在还不支持 patch 方法

 I checked the related functions, and currently the fetch method does not support the patch method.

**Reply by alexmathew11** [reply]
Wow a well known developer melian miko is also asking help. Nice 👍. 

And melian miko you should also tell that it's happening in mi band 7 which is not officially a Zepp os device

**Reply by melianmiko** [reply]
> And melian miko you should also tell that it's happening in mi band 7 which is not officially a Zepp os device

This is a bug of Zepp application and Side-Service API, this isn't Mi Band 7 specific.

**Comment by htoooth** [comment]
‌‌‌‌The patch method is supported in the latest version of the App.  @melianmiko
